In [4]:
import pandas as pd

In [22]:
df = pd.read_excel("../data/Assignment for a Data Manager Position in Gifka App.xlsx", sheet_name="Gift Card #3 - Raaionit")

In [23]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 347 entries, 0 to 346
Data columns (total 7 columns):
 #   Column                                                                   Non-Null Count  Dtype 
---  ------                                                                   --------------  ----- 
 0   Card Name                                                                347 non-null    object
 1   Card WebPage - (it seems like the site is currently under construction)  347 non-null    object
 2   Brand Name + Branch Name                                                 347 non-null    object
 3   Branch Address                                                           347 non-null    object
 4   Branch Phone Number                                                      345 non-null    object
 5   Brand Website                                                            329 non-null    object
 6   Brand Description                                                        237 non-nu

In [26]:
unique_brand_names = df['Brand Name + Branch Name'].unique()
unique_brand_names

array(['מקס ברנר כרמיאל', 'מקס ברנר ירושלים', 'SAI SUSHI BAR',
       "ג'אפן ג'אפן טבריה", "ג'אפן ג'אפן יקנעם עילית",
       "ג'אפן ג'אפן אור עקיבא", "ג'פאן ג'פאן חדרה", "ג'אפן ג'אפן בית שמש",
       "ג'אפן ג'אפן אריאל", "ג'פאן ג'פאן פתח תקווה",
       "ג'אפן ג'אפן תל אביב", "ג'אפן ג'אפן גבעת שמואל",
       "ג'אפן ג'אפן ירושלים", "ג'אפן ג'אפן מצפה רמון", "ג'ויה תל אביב",
       "באשר פרומז'רי קריית מוצקין", "באשר פרומז'רי חיפה",
       "באשר פרומז'רי עפולה", "באשר פרומז'רי זכרון יעקב",
       "באשר פרומז'רי נתניה", "באשר פרומז'רי הוד השרון",
       "באשר פרומז'רי רעננה", "באשר פרומז'רי תל אביב",
       "באשר פרומז'רי סניף גבעתיים",
       "באשר פרומז'רי ראשון לציון - הסניף סגור עד להודעה חדשה - עמכם הסליחה!",
       "באשר פרומז'רי מודיעין", "באשר פרומז'רי רחובות",
       "באשר פרומז'רי מישור אדומים", "באשר פרומז'רי מעלה אדומים",
       "באשר פרומז'רי ירושלים",
       "באשר פרומז'רי אשדוד - הסניף סגור עד להודעה חדשה - עמכם הסליחה!",
       "באשר פרומז'רי נתיבות", "באשר פרומז'רי באר שבע"

In [28]:
no_of_unique_brand_names = df['Brand Name + Branch Name'].nunique()
no_of_unique_brand_names 

347

As mentioned in the instructions that the unique brands are just 187 however total entries are 347.

As per in the hebrew the unique entries are given as the total number of entries. 

Therefore, going for language translation.

In [35]:
from googletrans import Translator
def translate_column_hebrew_to_english(text : str) -> str:
    translated = Translator().translate(text, src='he', dest='en')
    return translated.text

In [36]:
translated_text = translate_column_hebrew_to_english(text= "מקס ברנר כרמיאל")
translated_text

AttributeError: 'NoneType' object has no attribute 'group'

In [47]:
translator = Translator()
translator.translate(text ='Hallo', dest= 'en', src='de')
translator.text

AttributeError: 'NoneType' object has no attribute 'group'